### PR TITLE
Add `EnumMeta.__new__` & `EnumMeta.__call__`

### LINKED ISSUE #4324: Type check does not support Enum function api
Hi, there,

I am facing warnings of type checks when using function call of Enum.

You can take a look at this mypy playground.
https://mypy-play.net/?mypy=latest&python=3.8&gist=39161e33d3466390d6bf30a42e4615f7

My peer guesses this is a limitation of the stub in typeshed at
here https://github.com/python/typeshed/blob/master/stdlib/3/enum.pyi#L23
Not sure how to type this.

```
# Enum functional call documentation:
# https://docs.python.org/3/library/enum.html#functional-api

fake_enum1 = Enum('fake_enum1', ['a', 'b'])
fake_enum2 = Enum('fake_enum1', names=['a', 'b'])
fake_enum3 = Enum(value='fake_enum1', names=['a', 'b'])

fake_enum4 = Enum(value='fake_enum1', names=['a', 'b'] , module=__name__)


# main.py:7: error: Unexpected arguments to Enum()
# main.py:8: error: Unexpected arguments to Enum()
# main.py:10: error: Too many arguments for Enum()
# Found 3 errors in 1 file (checked 1 source file)
```

### PR DESCRIPTION
**Pull request by AlexWaygood**
Closes #4324, formally acknowledging in the stub that there is an alternative functional API for creating a new `Enum` class.

From testing this patch, locally, it seems that this has no effect on mypy's special-casing of `Enum`s:

```python
from enum import Enum

class Menu(Enum):
    SPAM = 1
    HAM = 2

reveal_type(Menu)                              # Revealed type is "def (value: builtins.object) -> test_enum.Menu*"
reveal_type(Menu.SPAM)                         # Revealed type is "Literal[test_enum.Menu.SPAM]?"
reveal_type(Menu(1))                           # Revealed type is "test_enum.Menu*"

Colours = Enum('Colours', 'RED YELLOW BLUE')
reveal_type(Colours)                           # Revealed type is "def (value: builtins.object) -> test_enum.Colours*"
reveal_type(Colours.RED)                       # Revealed type is "Literal[test_enum.Colours.RED]?"
reveal_type(Colours(1))                        # Revealed type is "test_enum.Colours*"
```

The command I ran to do this test was

```
mypy --python-version 3.10 --no-site-packages --show-traceback --no-implicit-optional --disallow-any-generics --warn-incomplete-stub --no-error-summary --custom-typeshed-dir . --warn-unused-ignores test_enum.py
```

Cc: @sobolevn (I was already playing around with the `Enum` stub before you submitted your patch 😀)

### CODE REVIEW

**Comment by AlexWaygood** [comment]
@erictraut, would it be possible to switch off pyright's `reportSelfClsParameterName` check for metaclasses? I could just add `# type: ignore`, but the check doesn't really make much sense when it comes to metaclasses.

**Comment by erictraut** [comment]
@AlexWaygood, I'd entertain a discussion on whether to disable this in specific cases if it makes sense, but it probably makes sense to do so in the context of an issue in the pyright issue tracker. If you want to have that discussion, please open an issue there. There are a bunch of questions we'd need to consider such as 1) what constitutes a metaclass; is it any class that derives from `type`?, 2) is this check suppressed for all methods in a metaclass or only certain dundered methods?, 3) is there some standard parameter name we'd want to enforce in those cases?

I recommend adding a `# type: ignore` for now.

**Comment by AlexWaygood** [comment]
> There are a bunch of questions we'd need to consider

I figured, but thought I'd mention it anyway 🙂 Thanks! I'll add a `# type: ignore` for now.

**Review by sobolevn — APPROVED** [review]
Awesome! Thanks for fixing it!

File: stdlib/enum.pyi [hunk 1]
```diff
@@ -2,24 +2,86 @@ import sys
 import types
 from abc import ABCMeta
 from builtins import property as _builtins_property
-from typing import Any, Iterator, Type, TypeVar
+from typing import Any, Dict, Iterable, Iterator, Mapping, Tuple, Type, TypeVar, Union, overload
```
Comment (srittau) [inline stdlib/enum.pyi:5]: Nit: `Iterable`, `Iterator`, and `Mapping` should be imported from `collections.abc`.

Comment (AlexWaygood) [reply stdlib/enum.pyi]: Done! 🙂

**Review by srittau — APPROVED** [review]

**Pull request merged**